src/editors/select.js

```diff
@@ -39,7 +39,7 @@ export var SelectEditor = AbstractEditor.extend({
     return Math.min(12,Math.max(longest_text/7,2));
   },
   typecast: function(value) {
-    if (this.schema.type === "boolean") return !!value;
+    if (this.schema.type === "boolean") return value == 'undefined' || value === undefined ? undefined : !!value;  
```

Comment:
Can't we do `Boolean(value);` here?

Reply:
`!!value` will convert `value` to Boolean, and is better to use than the `Boolean()` function.
But the problem is the undefined value. In some cases it's a string and in other cases it's undefined. So I can't see any other way to do it.

Reply:
> !!value will convert value to Boolean, and is better to use than the Boolean() function.

Why? `Boolean()` is much clearer.

> But the problem is the undefined value. In some cases it's a string and in other cases it's undefined. So I can't see any other way to do it.

So a string "undefined" would be true? and an `undefined` values would be `undefined`?
Actually boolean values should either be `true` or `false` ... is this related to optional properties?

Reply:
> 
> 
> > !!value will convert value to Boolean, and is better to use than the Boolean() function.
> 
> Why? `Boolean()` is much clearer.

http://adripofjavascript.com/blog/drips/the-difference-between-boolean-objects-and-boolean-primitives-in-javascript.html

> 
> > But the problem is the undefined value. In some cases it's a string and in other cases it's undefined. So I can't see any other way to do it.
> 
> So a string "undefined" would be true? and an `undefined` values would be `undefined`?

`"undefined"` evaluates to **TRUE**, but `undefined` evaluates to **FALSE**. Both are wrong as the value should stay `undefined` 

> Actually boolean values should either be `true` or `false` ... is this related to optional properties?

Did you read the text I put in the PR? It happens when the property is NOT required.

Reply:
@pmk65 Thank you for the explanations! :+1: